README.md

```diff
@@ -117,47 +117,117 @@ If you'd like to get a message when the button is pressed or released, call the
+more clear in the example below.
+
+The following shows an example MCP3008 Analog to Digital Converter (ADC) that reads from 
+an analog moisture sensor, converts the analog measurements to digital, and 
+sends the digital measurements to SPI pins on a Raspberry Pi.  An ADC is 
+used to convert the analog signal from a sensor to a digital signal, so the 
+signal can be read by the main processor running Linux (e.g. Raspberry Pi.) 
+A Raspberry Pi can't read an analog signal, so it needs an ADC to convert the
+signal to digital.
+
+UPDATED SCHEMATIC?
 
 ![SPI schematic](assets/images/schematic-adc.png)
 
-The protocol for talking to the ADC is described in the [MCP3202
-Datasheet](http://www.microchip.com/wwwproducts/Devices.aspx?dDocName=en010532).
-Sending a 0x64 first reads the temperature and sending a 0x74 reads the
-potentiometer.
+The protocol for talking to the ADC can be found on the MCP3008 data sheet 
+which you will find on the [MCP3008 product page](http://www.microchip.com/wwwproducts/en/MCP3008).  If 
+you are new to hardware, a data sheet can be intimidating but they become
+much easier to read once you know what you are looking for.  The protocol
+is very similar to an application program interface (API) in the software world.  
+
+In the case of the MCP3008, the protocol is found in section 5.0 "SERIAL
+COMMUNICATION" on page 19.  The protocol will tell you the position and 
+function of the bits you will send to the ADC, along with how the data (in the
+form of bits) will be returned. 
+
+Figure 5-1 on pg 20 shows 4 lines, CS (chip select), CLK (clock), DIN (data in), and 
+DOUT (data out).  We'll look at the DIN line first to determine what to send.  Then 
+we'll look at the DOUT line to determine what we will receive.
+
+TODO INSERT & DESCRIBE HOW TO READ FIGURE 5-1, SEE JOY OF HARDWARE SLIDES
+
+Using the protocol from the data sheet here is how we will read the moisture 
+sensor.
 
     # Make sure that you've enabled or loaded the SPI driver or this will
     # fail.
     iex> alias ElixirALE.SPI
     iex> {:ok, pid} = SPI.start_link("spidev0.0")
     {:ok, #PID<0.124.0>}
 
-    # Read the potentiometer
+    # Read the moisture sensor
+
+    # Use binary pattern matching to pull out the ADC counts (low 10 bits)
+    iex> mode = 1 #Single-ended mode
+    iex> sensor = 2 #Try the CH2 sensor
+    iex> <<_::size(14), value::size(10)>> =
+           SPI.transfer(pid, <<0x01, mode::size(1), sensor::size(3), 0::size(12)>>)
+    "NEED WHAT IT RETURNS"
+
+    iex> IO.puts "Sensor #{sensor} measured #{value}"
+    "NEED WHAT IT RETURNS"
+
+If you are new to using Elixir for embedded, there are a couple things here you 
+might not be familiar with.  We'll be working with bit strings.  If this is 
+new to you, you might want to refer to the 
+[Elixir docs for bitstings](https://hexdocs.pm/elixir/Kernel.SpecialForms.html#%3C%3C%3E%3E/1).
+
+TODO DISCUSS START LINK...WHY "spidev0.0"?
```

Comment:
Elixir ALE uses the Linux `spidev` device driver for communicating with SPI devices. It creates files in `/dev` for each SPI interface that it knows about. Many boards only have one SPI interface and it shows up as `/dev/spidev0.0`. The easiest way to find out the right name is to run `ls spidev*` in the `/dev` directory. If nothing shows up, it is likely that the `spidev` driver needs to be loaded. Try running `modprobe spidev` or consult your board's documentation.